### PR TITLE
Dev/foundry

### PR DESCRIPTION
**Pull request by DhairyaSethi**
Set up Foundry as the framework for compilation, deployment & testing. 
Lint & Slither builds will be fixed in subsequent [PRs](https://github.com/DhairyaSethi/fx-portal-contracts/pull/2).

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -4,22 +4,40 @@ on:
...
+      - name: Install Foundry
+        uses: foundry-rs/foundry-toolchain@v1
+        with:
+          version: nightly
+      - name: Build
+        run: forge build
+      - name: Run tests
+        run: forge test -vvv
+      - name: Get coverage
+        run: forge coverage
```
Comment (ZeroEkkusu) [inline .github/workflows/ci.yml]: Not needed since we have a coverage job below.

```suggestion
```

File: remappings.txt
Comment (ZeroEkkusu) [inline remappings.txt:1]: I've never used `remappings.txt` - do we need it for something?

File: test/FxRoot.t.sol [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        vm.expectRevert();
+        fxRoot.setFxChild(makeAddr("rand"));
+    }
+
+    function testSendMessageToChild() public {
+        vm.startPrank(alice);
+        MockMessageProcessor mockTunnel = new MockMessageProcessor();
+        bytes memory data = new bytes(0);
+
+        vm.expectEmit(true, false, false, true, address(mockTunnel));
```
Comment (ZeroEkkusu) [inline test/FxRoot.t.sol]: Btw, I prefer `expectEmit(address)` when possible, so all topics are checked.

File: remappings.txt
Comment (QEDK) [reply remappings.txt:1]: The forge docs recommend it for VS Code, not sure what it does.

Comment (DhairyaSethi) [reply remappings.txt:1]: yep^, it's for the solidity vs code intellisense for packages added using forge; most straightforward approach after manually adding all through extension settings. more context [here](https://github.com/foundry-rs/book/pull/466)

File: test/FxRoot.t.sol (hunk 2 above)
Comment (DhairyaSethi) [reply test/FxRoot.t.sol]: oh, overlooked this overload. yep this does check all topics as there's only one indexed topic but yeah more readable this way. [a21b21b](https://github.com/0xPolygon/fx-portal/pull/53/commits/a21b21bbc59ffb88ab14264d180b374f31d271b4)

File: remappings.txt
Comment (ZeroEkkusu) [reply remappings.txt:1]: I think Nomic Foundation's Solidity extension takes care of the remappings, but we can leave the file if you prefer it.

**Review by ZeroEkkusu — APPROVED** [review]

**Pull request merged**